### PR TITLE
meta: updated repo structure

### PR DESCRIPTION
**Pull request by ovflowd**
This PR updates relevant sections of the `README`, `CONTRIBUTING.md` and `COLLABORATOR_GUIDE.md` with updated guides, references and technologies.

cc @nodejs/website

### CODE REVIEW

**Comment by ovflowd** [comment]
cc @nodejs/website as we need a full review here! (As this PR updates the Contributing Guidelines)

File: COLLABORATOR_GUIDE.md [hunk 1]
```diff
@@ -58,21 +58,20 @@ full responsibility for the change, defer to another Collaborator.
...
-72 hours over weekends to account for international time differences
-and work schedules. Trivial changes (e.g. those which fix minor bugs
-or improve performance without affecting API or causing other
-wide-reaching impact) may be landed after a shorter delay. Any press
-release can land with no time constraints as long as the copy is
-properly formatted, it is not the responsibility of the website group
-to review the copy itself.
+Seventy-two hours over weekends to account for international time differences
+and work schedules. Trivial changes (e.g., those which fix minor bugs
+or improve performance without affecting API or causing another wide-reaching impact)
```
Comment (mikeesto) [inline COLLABORATOR_GUIDE.md]:
```suggestion
or improve performance without causing wide-reaching impact)
```

File: COLLABORATOR_GUIDE.md [hunk 2]
```diff
@@ -32,21 +33,20 @@ long as this is done while being mindful of these guidelines, the
...
 Collaborators may **close** any issue or pull request they believe is
-not relevant for the future of the Node.js project. Where this is
-unclear, the issue should be left open for several days to allow for
+not relevant to the future of the Node.js project. Where this is
+unclear, the issue should be left open for several days for
 additional discussion. Where this does not yield input from Node.js
-Collaborators or additional evidence that the issue has relevance, the
-issue may be closed. Remember that issues can always be re-opened if
-necessary.
+Collaborators or additional evidence that the issue has relevance may be closed.
```
Comment (mikeesto) [inline COLLABORATOR_GUIDE.md]:
```suggestion
Collaborators or additional evidence that the issue has relevance, then the issue may be closed.
```

**Review by mikeesto — APPROVED** [review]
LGTM, just made a couple of small suggestions

File: COLLABORATOR_GUIDE.md [hunk 3]
```diff
@@ -87,12 +86,54 @@ where a pull request:
...
+The Website also uses several other Open Source libraries (not limited to) listed below:
+
+- Styling is done with [SCSS][] and CSS Modules
+- [TailwindCSS](https://tailwindcss.com/) is used as our CSS Framework and the Foundation of our Design System
+- [Hero Icons](https://heroicons.com/) is an SVG Icon Library used within our Codebase
+- [Shiki](https://github.com/shikijs/shiki) is a Syntax Highlighter used for our Codeboxes
+- [MDX][] and Markdown are used for structuring the Content of the Website
+- [`react-intl`][] is the i18n Library adopted within the Website
+- [`next-sitemap`](https://www.npmjs.com/package/next-sitemap) is used for Sitemap and `robots.txt` Generation
+- We use [Rehype](https://github.com/rehypejs/rehype) and [Remark](https://github.com/remarkjs/remark) to extend MDX functionality
```
Comment (bmuenzenmeyer) [inline COLLABORATOR_GUIDE.md:107]: nit: any reason to switch between styles of markdown links here?

File: COLLABORATOR_GUIDE.md [hunk 4]
```diff
@@ -87,12 +86,54 @@ where a pull request:
...
+  - Configuration for Locales is done within `/i18n/config.json`
+- Website Content is defined within `/pages`
+  - Initial development usually happens in English: `/pages/en`
+  - Localized versions of `/pages/en` are done within `/pages/{localeCode}`
+  - All content is in Markdown and is per locale.
+  - The top of each Markdown file is a YAML (Frontmatter) block for page-specific localization information passed to various templates.
+  - The bulk of the Markdown content for each page is referenced as `{children}` on their respective JSX Layout (`layouts/`)
+- Multi-Purpose React Hooks are defined on `/hooks`
+- Multi-Purpose TypeScript definitions are defined on `/types`
+- React Context Providers are defined on `/provides`
```
Comment (bmuenzenmeyer) [inline COLLABORATOR_GUIDE.md]:
```suggestion
- React Context Providers are defined on `/providers`
```

File: COLLABORATOR_GUIDE.md [hunk 5]
```diff
@@ -167,32 +213,37 @@ export default MyComponent;
...
 - Components should always be the `default` export of a React Component file
-- Avoid using DOM/Web APIs/`document`/`window` API access within a React Component. Use utilities or Hooks when you need a Reactive state
+- Avoid using DOM/Web APIs/`document`/`window` API access within a React Component.
+  Use utilities or Hooks when you need a Reactive state
 - Avoid making your Component too big. Deconstruct it into smaller Components/Hooks whenever possible
 
 ## Unit Tests and Storybooks
 
-Each new feature or bug fix should be accompanied by a unit test (when deemed valuable). We use [Jest][] as our test runner and [React Testing Library][] for our React unit tests.
+Each new feature or bug fix should accompany a unit test (when deemed valuable).
```
Comment (bmuenzenmeyer) [inline COLLABORATOR_GUIDE.md]: This change should be reverted. The rephrasing makes it sound like unit tests come first, and the features or bugs come along as a consequence of the test.

File: COLLABORATOR_GUIDE.md [hunk 6]
```diff
@@ -167,32 +213,37 @@ export default MyComponent;
...
-- Unit Tests should cover that the functionality of a given change is working as expected.
+- We recommend adding unit tests for content covering `util`, `scripts`, `hooks`, and `components` whenever possible.
+- Unit Tests should ensure that a given change's functionality is working as expected.
 - When creating unit tests for React components, we recommend that the tests cover all the possible states of the component.
-- We also recommend mocking external dependencies, if unsure about how to mock a certain dependency, raise the question on your Pull Request.
-  - We recommend using [Jest's Mock Functions](https://jestjs.io/docs/en/mock-functions) for mocking dependencies.
-  - We recommend using [Jest's Mock Modules](https://jestjs.io/docs/en/manual-mocks) for mocking dependencies that are not available on the Node.js runtime.
+- We also recommend mocking external dependencies, if unsure about how to mock a particular dependency, raise the question on your Pull Request.
+  - We recommend mocking dependencies by mocking [Jest's Mock Functions](https://jestjs.io/docs/en/mock-functions).
```
Comment (bmuenzenmeyer) [inline COLLABORATOR_GUIDE.md]: this sentence read better before

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -1,6 +1,6 @@
 # Node.js Website Contributing Guide
 
-Thank you for your interest in contributing to the Node.js Website. Before you proceed, briefly go through the following:
+Thank you for being so interested in contributing to the Node.js Website. Before you proceed, briefly go through the following:
```
Comment (bmuenzenmeyer) [inline CONTRIBUTING.md]: i think i flagged this one before too - it reads to my ears as patronizing and should be reverted

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -164,39 +170,41 @@ Commits should be signed. You can read more about [Commit Signing][] here.
...
     - The comment must mention `@nodejs/website` and must have at least one 👍 (or any other sort of approval reaction) if the person fast-tracking the PR is the author of the PR.
-    - Fast-tracking is only allowed for small bug fixes, small feature changes, localisation changes, or other sorts of non-critical/highly-impacting changes that are not covered by the previous rule that allows PRs to be merged immediately.
+    - Fast-tracking is only allowed for small bug fixes, small feature changes, localization changes, or other non-critical/highly-impacting changes not covered by the previous rule that allows PRs to be merged immediately.
     - Fast-tracking cannot be used for updates on the `COLLABORATOR_GUIDE.md`, `CONTRIBUTING.md` guide, `CODEOWNERS`, GitHub Actions, or any security-impacting file or document that changes the governing policies of this repository.
-- There must be no objections after a 48-hour period (Or 72 hours if the PR was authored on the weekend).
+- There must be no objections after 48 hours (Or 72 hours if the PR was authored on the weekend).
```
Comment (bmuenzenmeyer) [inline CONTRIBUTING.md]: nit: elsewhere in this same PR we change numerals to written out numbers

File: COLLABORATOR_GUIDE.md (hunk 3 above)
Comment (ovflowd) [reply COLLABORATOR_GUIDE.md:107]: We're using footnotes for any link used more than once.

File: COLLABORATOR_GUIDE.md (hunk 6 above)
Comment (ovflowd) [inline COLLABORATOR_GUIDE.md]:
```suggestion
  - We recommend using [Jest's Mock Functions](https://jestjs.io/docs/en/mock-functions) for mocking dependencies.
```

File: COLLABORATOR_GUIDE.md (hunk 5 above)
Comment (ovflowd) [inline COLLABORATOR_GUIDE.md]:
```suggestion
Each new feature or bug fix should be accompanied by a unit test (when deemed valuable).
```

File: CONTRIBUTING.md (hunk 7 above)
Comment (ovflowd) [inline CONTRIBUTING.md]:
```suggestion
Thank you for your interest in contributing to the Node.js Website. Before you proceed, briefly go through the following:
```

File: CONTRIBUTING.md (hunk 8 above)
Comment (ovflowd) [inline CONTRIBUTING.md]:
```suggestion
- There must be no objections after forty-eight (48) hours (Or seventy-two (72) hours if the PR was authored on the weekend).
```

**Comment by ovflowd** [comment]
@bmuenzenmeyer I've addressed all comments!

File: COLLABORATOR_GUIDE.md (hunk 5 above)
Comment (bmuenzenmeyer) [reply COLLABORATOR_GUIDE.md]: apologies if i am not understanding something, but i dont think we should move forward with this language. if it was marked resolved, can you comment as to why?

File: COLLABORATOR_GUIDE.md (hunk 6 above)
Comment (bmuenzenmeyer) [reply COLLABORATOR_GUIDE.md]: `We recommend mocking dependencies by mocking Jest's Mock Functions` uses the word mock three times. unresolved this to consider reverting the phrasing

**Comment by bmuenzenmeyer** [comment]
@ovflowd I unresolved some comments. I am wondering if you intended to stage more changes? there are suggestions made by you that are not committed.

image

**Comment by ovflowd** [comment]
@bmuenzenmeyer can you double check you're not checking an old version of the diff? Everything is correct over here ;)

File: COLLABORATOR_GUIDE.md (hunk 5 above)
Comment (ovflowd) [reply COLLABORATOR_GUIDE.md]: I am not sure what you're talking about. I've applied the changes.

**Comment by bmuenzenmeyer** [comment]
> @bmuenzenmeyer can you double check you're not checking an old version of the diff? Everything is correct over here ;)

a refresh fixed it 😅

**Review by bmuenzenmeyer — APPROVED** [review]

File: COLLABORATOR_GUIDE.md (hunk 6 above)
Comment (ovflowd) [reply COLLABORATOR_GUIDE.md]: I'm re-resolving this, you're seeing a wrong version of the diff.

**Comment by ovflowd** [comment]
> > @bmuenzenmeyer can you double check you're not checking an old version of the diff? Everything is correct over here ;)
> 
> a refresh fixed it 😅

I was kinda getting confused lol

**Pull request merged**